champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/sword/Ride.java

```diff
@@ -87,22 +86,34 @@ public void activate(Player player, int level) {
         horse.addPassenger(player);
         HorseData data = new HorseData(horse, System.currentTimeMillis());
         horseData.put(player, data);
+    }
 
+    @UpdateEvent(delay = 500)
+    public void removeHorses() {
+        Set<Player> toRemove = new HashSet<>();
 
-        new BukkitRunnable() {
-            @Override
-            public void run() {
-                if (!horse.isDead()) {
-                    horse.remove();
-                    horseData.remove(player);
-                }
+        for (Map.Entry<Player, HorseData> data : horseData.entrySet()) {
+            Player player = data.getKey();
+            int level = getLevel(player);
```

Comment:
Instead of getting a players skill level every half second, you could also store the lifespan in HorseData